spec/migrations/20241017013023_reencrypt_password_scramsha_spec.rb

```diff
@@ -23,9 +23,9 @@
 
       username = ActiveRecord::Base.connection_db_config.configuration_hash[:username]
 
-      expect(ActiveRecord::Base.connection_db_config).to receive(:configuration_hash).exactly(10).times.and_call_original
-      expect(ActiveRecord::Base.connection_db_config).to receive(:configuration_hash).and_wrap_original do |original_method, *args, &block|
-        original_method.call(*args, &block).dup.tap { |i| i.delete(:password) }
+      allow(ActiveRecord::Base.connection_db_config).to receive(:configuration_hash).and_wrap_original do |original_method, *args, &block|
+        # set a value for any calls originating from the migration file, not from rails itself
+        original_method.call(*args, &block).dup.tap { |i| i.delete(:password) if caller_locations.any? {|loc| loc.path.include?("db/migrate/20241017013023_reencrypt_password_scramsha.rb")} }
```

Comment:
Note, this is still super brittle but that's because the test is modifying the database configuration in the context of the test and code outside our migration can be impacted by that change.  This is why we had 10 calls originally in rails 7.0 and only 1 in 7.1.